### ISSUE TITLE
Throw `AnalysisException` where declared class used `= with mixin` syntax.

### ISSUE THREAD

**Issue by zhengxiaoyao0716** [dart-lang/json_serializable#387]
When I use the `class Xxx = Yyy with Zzz` syntax, then the build would failed and throw an exception.
Here the example code:
``` dart
import 'package:json_annotation/json_annotation.dart';
import 'package:json_test/alias.dart';
import 'package:meta/meta.dart';

part 'config.g.dart'; // flutter packages pub run build_runner build

// This way, all works well.
// class Url extends Alias<String> {
//   Url(String raw) : super(raw);
// }

// This usage would caused `AnalysisException: Cannot compute LIBRARY_ELEMENT for Instance of 'AssetBasedSource'`
class Url = Alias<String> with Type;
// But if we cleared the `.dart_tool` folder and rebuild the `config.g.dart`,
// it would works succeed until we changed the code and build them again.

@JsonSerializable()
class Config {
  static Config init = Config(locale: 'zh', servers: []);

  String locale;
  @JsonKey(fromJson: _serversFromJson, toJson: _serversToJson)
  final List<Url> servers;

  Config({
    @required this.locale,
    @required this.servers,
  });

  static List<Url> _serversFromJson(List<String> list) =>
      list.map((url) => Url(url)).toList();
  static List<String> _serversToJson(List<Url> servers) =>
      servers.map((url) => url.raw).toList();

  factory Config.fromJson(Map<String, dynamic> json) => _$ConfigFromJson(json);
  Map<String, dynamic> toJson() => _$ConfigToJson(this);
}

```

***
> `pubspec.yaml`
``` yaml
dev_dependencies:
  test: ^1.0.0
  build_runner: ^1.0.0
  json_serializable: ^2.0.0
```

> `flutter --version`
```
Flutter 1.0.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 5391447fae (3 months ago) • 2018-11-29 19:41:26 -0800
Engine • revision 7375a0f414
Tools • Dart 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)
```

**Comment by zhengxiaoyao0716** [comment]
Here the code of the `alias.dart`, I used it make a way to quickly simulated the `alias` syntax, because of I have a series of `String` expressed different meanings, like `Id`, `Name`, `Url` and so on.
```
class Boxed<T> {
  final T raw;

  const Boxed(this.raw);

  Boxed.of(Boxed<T> other) : this(other.raw);

  @override
  String toString() {
    return raw.toString();
  }

  @override
  bool operator ==(other) =>
      other.runtimeType == this.runtimeType && other.raw == raw;

  @override
  int get hashCode => raw.hashCode;
}

mixin Alias {}

```

**Comment by zhengxiaoyao0716** [comment]
Here the complete exception report:
``` log
[INFO] Running build...
[SEVERE] json_serializable on lib/config.dart:

AnalysisException: Cannot compute LIBRARY_ELEMENT for Instance of 'AssetBasedSource'
Caused by Unexpected exception while performing ResolveUnitTypeNamesTask for source json_test|lib/config.dart
#0      AnalysisTask._safelyPerform (package:analyzer/src/task/api/model.dart:353:7)
#1      AnalysisTask.perform (package:analyzer/src/task/api/model.dart:240:7)
#2      AnalysisDriver.performWorkItem (package:analyzer/src/task/driver.dart:285:10)
#3      AnalysisDriver.computeResult (package:analyzer/src/task/driver.dart:110:22)
#4      AnalysisContextImpl.computeResult (package:analyzer/src/context/context.dart:654:14)
#5      AnalysisContextImpl.computeLibraryElement (package:analyzer/src/context/context.dart:622:12)
#6      AnalyzerResolver.libraryFor (package:build_resolvers/src/resolver.dart:114:21)
<asynchronous suspension>
#7      PerActionResolver.libraryFor (package:build_resolvers/src/resolver.dart:65:17)
#8      _DelayedResolver.libraryFor (package:build/src/builder/build_step_impl.dart:192:25)
<asynchronous suspension>
#9      BuildStepImpl.inputLibrary (package:build/src/builder/build_step_impl.dart:37:21)
<asynchronous suspension>
#10     _Builder.build (package:source_gen/src/builder.dart:65:31)
<asynchronous suspension>
#11     runBuilder.buildForInput (package:build/src/generate/run_builder.dart:45:21)
<asynchronous suspension>
#12     MappedListIterable.elementAt (dart:_internal/iterable.dart:414:29)
#13     ListIterator.moveNext (dart:_internal/iterable.dart:343:26)
#14     Future.wait (dart:async/future.dart:392:26)
#15     runBuilder.<anonymous closure> (package:build/src/generate/run_builder.dart:51:36)
#16     _rootRun (dart:async/zone.dart:1124:13)
#17     _CustomZone.run (dart:async/zone.dart:1021:19)
#18     _runZoned (dart:async/zone.dart:1516:10)
#19     runZoned (dart:async/zone.dart:1500:12)
#20     scopeLogAsync (package:build/src/builder/logging.dart:22:3)
#21     runBuilder (package:build/src/generate/run_builder.dart:51:9)
<asynchronous suspension>
#22     _SingleBuild._runForInput.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:455:19)
#23     _NoOpBuilderActionTracker.trackStage (package:build_runner_core/src/generate/performance_tracker.dart:303:15)
#24     _SingleBuild._runForInput.<anonymous closure>.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:453:23)
<asynchronous suspension>
#25     NoOpTimeTracker.track (package:timing/src/timing.dart:222:44)
#26     _SingleBuild._runForInput.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:412:22)
#27     Pool.withResource (package:pool/pool.dart:127:28)
<asynchronous suspension>
#28     _SingleBuild._runForInput (package:build_runner_core/src/generate/build_impl.dart:408:17)
#29     _SingleBuild._runBuilder.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:358:38)
#30     MappedIterator.moveNext (dart:_internal/iterable.dart:392:18)
#31     Future.wait (dart:async/future.dart:392:26)
#32     _SingleBuild._runBuilder (package:build_runner_core/src/generate/build_impl.dart:357:36)
#33     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#34     _SingleBuild._runBuilder (package:build_runner_core/src/generate/build_impl.dart:355:40)
#35     _SingleBuild._runPhases.<anonymous closure>.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:304:20)
#36     _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#37     _rootRunUnary (dart:async/zone.dart:1132:38)
#38     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#39     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#40     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#41     Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#42     Future._complete (dart:async/future_impl.dart:476:7)
#43     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#44     _AsyncAwaitCompleter.complete (dart:async/runtime/libasync_patch.dart:28:18)
#45     _completeOnAsyncReturn (dart:async/runtime/libasync_patch.dart:295:13)
#46     _SingleBuild._matchingPrimaryInputs (package:build_runner_core/src/generate/build_impl.dart)
#47     _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#48     _rootRunUnary (dart:async/zone.dart:1132:38)
#49     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#50     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#51     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#52     Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#53     Future._completeWithValue (dart:async/future_impl.dart:486:5)
#54     Future.wait.<anonymous closure> (dart:async/future.dart:399:22)
#55     _rootRunUnary (dart:async/zone.dart:1132:38)
#56     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#57     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#58     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#59     Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#60     Future._complete (dart:async/future_impl.dart:476:7)
#61     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#62     _AsyncAwaitCompleter.complete.<anonymous closure> (dart:async/runtime/libasync_patch.dart:33:20)
#63     _rootRun (dart:async/zone.dart:1120:38)
#64     _CustomZone.run (dart:async/zone.dart:1021:19)
#65     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#66     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#67     _rootRun (dart:async/zone.dart:1124:13)
#68     _CustomZone.run (dart:async/zone.dart:1021:19)
#69     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#70     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#71     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#72     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#73     _runPendingImmediateCallback (dart:isolate/runtime/libisolate_patch.dart:115:13)
#74     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:172:5)

Caused by NoSuchMethodError: The method 'substitute2' was called on null.
Receiver: null
Tried calling: substitute2(_List len:1, _List len:1)
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
#1      ClassElementImpl._computeMixinAppConstructors.<anonymous closure> (package:analyzer/src/dart/element/element.dart:1140:35)
#2      MappedIterator.moveNext (dart:_internal/iterable.dart:392:18)
#3      new List.from (dart:core/runtime/libarray_patch.dart:40:17)
#4      Iterable.toList (dart:core/iterable.dart:398:16)
#5      ClassElementImpl._computeMixinAppConstructors (package:analyzer/src/dart/element/element.dart:1147:8)
#6      ClassElementImpl.constructors (package:analyzer/src/dart/element/element.dart:518:30)
#7      ClassElementImpl.unnamedConstructor (package:analyzer/src/dart/element/element.dart:949:40)
#8      InterfaceTypeImpl.lookUpConstructor (package:analyzer/src/dart/element/type.dart:1881:36)
#9      AstRewriteVisitor.visitMethodInvocation (package:analyzer/src/generated/resolver.dart:92:18)
#10     MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:8679:49)
#11     ExpressionFunctionBodyImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:4189:18)
#12     UnifyingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:3440:10)
#13     UnifyingAstVisitor.visitExpressionFunctionBody (package:analyzer/dart/ast/visitor.dart:3288:65)
#14     ExpressionFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:4185:15)
#15     FunctionExpressionImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:5887:12)
#16     UnifyingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:3440:10)
#17     UnifyingAstVisitor.visitFunctionExpression (package:analyzer/dart/ast/visitor.dart:3337:57)
#18     ScopedVisitor.visitFunctionExpression (package:analyzer/src/generated/resolver.dart:6461:15)
#19     FunctionExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:5881:49)
#20     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9174:20)
#21     ArgumentListImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:423:16)
#22     UnifyingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:3440:10)
#23     UnifyingAstVisitor.visitArgumentList (package:analyzer/dart/ast/visitor.dart:3178:45)
#24     ArgumentListImpl.accept (package:analyzer/src/dart/ast/ast.dart:419:49)
#25     MethodInvocationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:8686:20)
#26     UnifyingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:3440:10)
#27     UnifyingAstVisitor.visitMethodInvocation (package:analyzer/dart/ast/visitor.dart:3425:53)
#28     AstRewriteVisitor.visitMethodInvocation (package:analyzer/src/generated/resolver.dart:65:11)
#29     MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:8679:49)
#30     MethodInvocationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:8683:14)
#31     UnifyingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:3440:10)
#32     UnifyingAstVisitor.visitMethodInvocation (package:analyzer/dart/ast/visitor.dart:3425:53)
#33     AstRewriteVisitor.visitMethodInvocation (package:analyzer/src/generated/resolver.dart:65:11)
#34     MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:8679:49)
#35     ExpressionFunctionBodyImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:4189:18)
#36     UnifyingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:3440:10)
#37     UnifyingAstVisitor.visitExpressionFunctionBody (package:analyzer/dart/ast/visitor.dart:3288:65)
#38     ExpressionFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:4185:15)
#39     MethodDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:8541:12)
#40     UnifyingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:3440:10)
#41     UnifyingAstVisitor.visitMethodDeclaration (package:analyzer/dart/ast/visitor.dart:3422:55)
#42     ScopedVisitor.visitMethodDeclarationInScope (package:analyzer/src/generated/resolver.dart:6597:11)
#43     ScopedVisitor.visitMethodDeclaration (package:analyzer/src/generated/resolver.dart:6590:7)
#44     MethodDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:8532:49)
#45     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9174:20)
#46     ScopedVisitor.visitClassMembersInScope (package:analyzer/src/generated/resolver.dart:6218:18)
#47     ScopedVisitor.visitClassDeclaration (package:analyzer/src/generated/resolver.dart:6196:11)
#48     ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:1765:49)
#49     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9174:20)
#50     CompilationUnitImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2567:21)
#51     UnifyingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:3440:10)
#52     UnifyingAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:3235:51)
#53     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2560:49)
#54     ResolveUnitTypeNamesTask.internalPerform (package:analyzer/src/task/dart.dart:5133:10)
#55     AnalysisTask._safelyPerform (package:analyzer/src/task/api/model.dart:341:9)
#56     AnalysisTask.perform (package:analyzer/src/task/api/model.dart:240:7)
#57     AnalysisDriver.performWorkItem (package:analyzer/src/task/driver.dart:285:10)
#58     AnalysisDriver.computeResult (package:analyzer/src/task/driver.dart:110:22)
#59     AnalysisContextImpl.computeResult (package:analyzer/src/context/context.dart:654:14)
#60     AnalysisContextImpl.computeLibraryElement (package:analyzer/src/context/context.dart:622:12)
#61     AnalyzerResolver.libraryFor (package:build_resolvers/src/resolver.dart:114:21)
<asynchronous suspension>
#62     PerActionResolver.libraryFor (package:build_resolvers/src/resolver.dart:65:17)
#63     _DelayedResolver.libraryFor (package:build/src/builder/build_step_impl.dart:192:25)
<asynchronous suspension>
#64     BuildStepImpl.inputLibrary (package:build/src/builder/build_step_impl.dart:37:21)
<asynchronous suspension>
#65     _Builder.build (package:source_gen/src/builder.dart:65:31)
<asynchronous suspension>
#66     runBuilder.buildForInput (package:build/src/generate/run_builder.dart:45:21)
<asynchronous suspension>
#67     MappedListIterable.elementAt (dart:_internal/iterable.dart:414:29)
#68     ListIterator.moveNext (dart:_internal/iterable.dart:343:26)
#69     Future.wait (dart:async/future.dart:392:26)
#70     runBuilder.<anonymous closure> (package:build/src/generate/run_builder.dart:51:36)
#71     _rootRun (dart:async/zone.dart:1124:13)
#72     _CustomZone.run (dart:async/zone.dart:1021:19)
#73     _runZoned (dart:async/zone.dart:1516:10)
#74     runZoned (dart:async/zone.dart:1500:12)
#75     scopeLogAsync (package:build/src/builder/logging.dart:22:3)
#76     runBuilder (package:build/src/generate/run_builder.dart:51:9)
<asynchronous suspension>
#77     _SingleBuild._runForInput.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:455:19)
#78     _NoOpBuilderActionTracker.trackStage (package:build_runner_core/src/generate/performance_tracker.dart:303:15)
#79     _SingleBuild._runForInput.<anonymous closure>.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:453:23)
<asynchronous suspension>
#80     NoOpTimeTracker.track (package:timing/src/timing.dart:222:44)
#81     _SingleBuild._runForInput.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:412:22)
#82     Pool.withResource (package:pool/pool.dart:127:28)
<asynchronous suspension>
#83     _SingleBuild._runForInput (package:build_runner_core/src/generate/build_impl.dart:408:17)
#84     _SingleBuild._runBuilder.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:358:38)
#85     MappedIterator.moveNext (dart:_internal/iterable.dart:392:18)
#86     Future.wait (dart:async/future.dart:392:26)
#87     _SingleBuild._runBuilder (package:build_runner_core/src/generate/build_impl.dart:357:36)
#88     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#89     _SingleBuild._runBuilder (package:build_runner_core/src/generate/build_impl.dart:355:40)
#90     _SingleBuild._runPhases.<anonymous closure>.<anonymous closure> (package:build_runner_core/src/generate/build_impl.dart:304:20)
#91     _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#92     _rootRunUnary (dart:async/zone.dart:1132:38)
#93     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#94     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#95     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#96     Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#97     Future._complete (dart:async/future_impl.dart:476:7)
#98     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#99     _AsyncAwaitCompleter.complete (dart:async/runtime/libasync_patch.dart:28:18)
#100    _completeOnAsyncReturn (dart:async/runtime/libasync_patch.dart:295:13)
#101    _SingleBuild._matchingPrimaryInputs (package:build_runner_core/src/generate/build_impl.dart)
#102    _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#103    _rootRunUnary (dart:async/zone.dart:1132:38)
#104    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#105    _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#106    Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#107    Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#108    Future._completeWithValue (dart:async/future_impl.dart:486:5)
#109    Future.wait.<anonymous closure> (dart:async/future.dart:399:22)
#110    _rootRunUnary (dart:async/zone.dart:1132:38)
#111    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#112    _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#113    Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#114    Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#115    Future._complete (dart:async/future_impl.dart:476:7)
#116    _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#117    _AsyncAwaitCompleter.complete.<anonymous closure> (dart:async/runtime/libasync_patch.dart:33:20)
#118    _rootRun (dart:async/zone.dart:1120:38)
#119    _CustomZone.run (dart:async/zone.dart:1021:19)
#120    _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#121    _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#122    _rootRun (dart:async/zone.dart:1124:13)
#123    _CustomZone.run (dart:async/zone.dart:1021:19)
#124    _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#125    _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#126    _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#127    _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#128    _runPendingImmediateCallback (dart:isolate/runtime/libisolate_patch.dart:115:13)
#129    _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:172:5)

```

**Comment by kevmoo** [comment]
I just ran this through and it worked fine. I have the latest `pkg:analyzer` at `0.35.2`.

What version of `package:analyzer` do you have? Look at `pubspec.lock`

**Comment by zhengxiaoyao0716** [comment]
![image](https://user-images.githubusercontent.com/12682063/53317744-dd305f80-3907-11e9-840d-47d7c811a375.png)

**Comment by zhengxiaoyao0716** [comment]
Have you tried change some code and build them again?
![image](https://user-images.githubusercontent.com/12682063/53318043-cdfde180-3908-11e9-9330-f5eb97f399b1.png)

**Comment by kevmoo** [comment]
I'm guessing this is a bug that was fixed in `0.35.x` of `package:analyzer`. I'm guessing you'll have to wait until Flutter upgrades their analyzer dependency to get the latest...

**Comment by zhengxiaoyao0716** [comment]
You are right, thank you.
